sdks/swift/Mentat/Mentat/Core/TypedValue.swift

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Mentatlib
+
+class TypedValue: OptionalRustObject {
+
+    var valueType: ValueType {
+        return typed_value_value_type(self.raw!)
+    }
+
+    func asLong() -> Int64 {
+        defer {
+            self.raw = nil
+        }
+        return typed_value_as_long(self.raw!)
```

Comment:
So if I call

```swift
logger.info("My value is \(tv.asLong())")
SomeAPI.getThing(tv.asLong())
```

my code is going to crash, no?

Reply:
Yes, just as it would if you tried to do that when getting a value twice from a result in `rusqlite` as I tried to replicate that behaviour. I was trying to make the the behaviour as similar as possible to what users might already have used, but clearly that has not worked as you are expecting it to behave differently. I will change these to be non-consuming.